src/test/java/com/thealgorithms/strings/AhoCorasickTest.java

```diff
@@ -0,0 +1,164 @@
+
+        // Perform the search using the Aho-Corasick algorithm
+        final var positions = AhoCorasick.search(text, patterns);
+
+        // Compare the actual results with the expected results
+        assertEquals(expected, positions);
+    }
+
+    /**
+     * Test searching with an empty pattern array.
+     * The result should be an empty map.
+     */
+    @Test
+    void testEmptyPatterns() {
+        // Define an empty pattern array
+        final var emptyPatterns = new String[] {};
+
+        // Perform the search using the Aho-Corasick algorithm
+        final var positions = AhoCorasick.search(text, emptyPatterns);
+
+        // Verify that no patterns are found (empty map)
+        assertTrue(positions.isEmpty());
+    }
+
+    /**
+     * Test searching for patterns that are not present in the input text.
+     * The result should be an empty list for each pattern.
+     */
+    @Test
+    void testPatternNotFound() {
+        // Define patterns that are not present in the text
+        final var searchPatterns = new String[] {"XYZ", "123"};
+
+        final var expected = Map.of("XYZ", new ArrayList<Integer>(), "123", new ArrayList<Integer>());
+
+        // Perform the search using the Aho-Corasick algorithm
+        final var positions = AhoCorasick.search(text, searchPatterns);
+
+        // Verify that empty lists are found for patterns that are not in the text
+        assertEquals(expected, positions);
+    }
+
+    /**
+     * Test searching for patterns that start at the beginning of the input text.
+     * The expected position for each pattern is 0.
+     */
+    @Test
+    void testPatternAtBeginning() {
+        // Define patterns that start at the beginning of the text
+        final var searchPatterns = new String[] {"GC", "GCA", "GCAT"};
+
+        // Define the expected positions (0) for each pattern
+        final var expected = Map.of("GC", new ArrayList<Integer>(Arrays.asList(0)), "GCA", new ArrayList<Integer>(Arrays.asList(0)), "GCAT", new ArrayList<Integer>(Arrays.asList(0)));
+        // Perform the search using the Aho-Corasick algorithm
+        final var positions = AhoCorasick.search(text, searchPatterns);
+
+        // Check that the first occurrence of each pattern is at the beginning (position 0)
+        assertEquals(expected, positions);
+    }
+
+    /**
+     * Test searching for patterns that end at the end of the input text.
+     * The expected positions are 4, 3, and 2 for the patterns.
+     */
+    @Test
+    void testPatternAtEnd() {
+        // Define patterns that end at the end of the text
+        final var searchPatterns = new String[] {"CG", "TCG", "ATCG"};
+
+        // Define the expected positions (4, 3, 2) for the patterns
+        final var expected = new int[] {4, 3, 2};
```

Comment:
Why not to define `expected` as `Map` as in other tests?

Reply:
got it, implemented it.